virttest/utils_misc.py

```diff
@@ -2170,24 +2170,30 @@ def _make_up_pattern(flags):
     if ARCH in ('ppc64', 'ppc64le'):
         return []     # remove -cpu and leave it on qemu to decide
 
-    cpu_types = {"AuthenticAMD": ["Opteron_G5", "Opteron_G4", "Opteron_G3",
-                                  "Opteron_G2", "Opteron_G1"],
-                 "GenuineIntel": ["Broadwell", "Broadwell-noTSX",
+    cpu_types = {"AuthenticAMD": ["EPYC", "Opteron_G5", "Opteron_G4",
+                                  "Opteron_G3", "Opteron_G2", "Opteron_G1"],
+                 "GenuineIntel": ["KnightsMill",
+                                  "Skylake-Server", "Skylake-Client",
+                                  "Broadwell", "Broadwell-noTSX",
                                   "Haswell", "Haswell-noTSX", "SandyBridge",
                                   "Westmere", "Nehalem",
                                   "Penryn", "Conroe"]}
-    cpu_type_re = {"Opteron_G5": "f16c,fma,tbm",
-                   "Opteron_G4":
-                   "avx,xsave,aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3,sse4a",
+    cpu_type_re = {"EPYC": "avx2,adx,bmi2,sha_ni",
+                   "Opteron_G5": "f16c,fma4,xop,tbm",
```

Comment:
suggest to change fma4 to fma, idea comes from the commit introduced cpu model G5:
commit 021941b9d97127bbd62a2ca39208d6a5d486b3cb
Author: Andre Przywara <osp@andrep.de>
Date:   Wed Nov 14 16:28:53 2012 -0200

    target-i386/cpu: Add new Opteron CPU model
    
    Add a new base CPU model called Opteron_G5 to model the latest
    Opteron CPUs. This increases the model value and model numbers and
    adds TBM, F16C and FMA over the latest G4 model.

Reply:
I see it is defined by qemu:
```c
    {
        .name = "Opteron_G5",
        .level = 0xd,
        .vendor = CPUID_VENDOR_AMD,
        .family = 21,
        .model = 2,
        .stepping = 0,
        .features[FEAT_1_EDX] =
            CPUID_VME | CPUID_SSE2 | CPUID_SSE | CPUID_FXSR | CPUID_MMX |
            CPUID_CLFLUSH | CPUID_PSE36 | CPUID_PAT | CPUID_CMOV | CPUID_MCA |
            CPUID_PGE | CPUID_MTRR | CPUID_SEP | CPUID_APIC | CPUID_CX8 |
            CPUID_MCE | CPUID_PAE | CPUID_MSR | CPUID_TSC | CPUID_PSE |
            CPUID_DE | CPUID_FP87,
        .features[FEAT_1_ECX] =
            CPUID_EXT_F16C | CPUID_EXT_AVX | CPUID_EXT_XSAVE |
            CPUID_EXT_AES | CPUID_EXT_POPCNT | CPUID_EXT_SSE42 |
            CPUID_EXT_SSE41 | CPUID_EXT_CX16 | CPUID_EXT_FMA |
            CPUID_EXT_SSSE3 | CPUID_EXT_PCLMULQDQ | CPUID_EXT_SSE3,
        /* Missing: CPUID_EXT2_RDTSCP */
        .features[FEAT_8000_0001_EDX] =
            CPUID_EXT2_LM | CPUID_EXT2_PDPE1GB | CPUID_EXT2_NX |
            CPUID_EXT2_SYSCALL,
        .features[FEAT_8000_0001_ECX] =
            CPUID_EXT3_TBM | CPUID_EXT3_FMA4 | CPUID_EXT3_XOP |    // ** at here
            CPUID_EXT3_3DNOWPREFETCH | CPUID_EXT3_MISALIGNSSE |
            CPUID_EXT3_SSE4A | CPUID_EXT3_ABM | CPUID_EXT3_SVM |
            CPUID_EXT3_LAHF_LM,
        /* no xsaveopt! */
        .xlevel = 0x8000001A,
        .model_id = "AMD Opteron 63xx class CPU",
    },
```

The reason of appending `xop` and `fma4` is to prevent `Opteron_G5` to be supported model for `EPYC` host. As `EPYC` supports `fma`, the change takes no advantage to distinguish the two different models.

I'd suggest to either remove or keep it, which one do you think is the best?

Reply:
Ok, keep fma4 here is ok for me